File: solutions/csharp/rotational-cipher/1/RotationalCipher.cs
using System;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        string cipher = "";
        int chi;
        foreach (char ch in text) {
            if (ch >= 'a' && ch <='z')
            {
                chi = 'a' + (ch + shiftKey - 'a') % 26;
                cipher += char.ConvertFromUtf32(chi);
            }
            else if (ch >= 'A' && ch <='Z') {
                chi = 'A' + (ch + shiftKey - 'A') % 26;
                cipher += char.ConvertFromUtf32(chi);
            }
            else cipher += char.ConvertFromUtf32(ch);
        }
        return cipher;
    }
}

File: solutions/java/rotational-cipher/1/src/main/java/RotationalCipher.java
class RotationalCipher {

    String alphabet = "abcdefghijklmnopqrstuvwxyz";
    String abrot;

    RotationalCipher(int shiftKey) {
        abrot="";
       for (int i = 0; i < 26; i++) {
            abrot += alphabet.substring((i + shiftKey) % 26, ((i + shiftKey) % 26) + 1);
        }
        //System.out.println("abrot="+abrot);
    }

    String rotate(String data) {
        String rot = "";
        for (int i = 0; i < data.length(); i++) {
            if (alphabet.indexOf(data.substring(i, i + 1).toLowerCase()) >= 0) {
                if (Character.isUpperCase(data.charAt(i))) {
                    rot += abrot.substring(alphabet.indexOf(data.substring(i, i + 1).toLowerCase()),
                            alphabet.indexOf(data.substring(i, i + 1).toLowerCase()) + 1).toUpperCase();
                } else {
                    rot += abrot.substring(alphabet.indexOf(data.substring(i, i + 1)),
                            alphabet.indexOf(data.substring(i, i + 1)) + 1);
                }
            } else {
                rot += data.substring(i, i + 1);
            }
            //System.out.println("rot="+rot);
        }
        return(rot);
    }

}